test/test.js

```diff
@@ -255,4 +264,41 @@ describe('request-language', function() {
     };
     expect(method).to.throw('You haven\'t defined the markup `{language}` in your cookie.url settings.');
   });
+
+  it('should set the language requested in the query string', function() {
+    var middleware = requestLangauge({
+      languages: ['en-US', 'zh-CN'],
+      cookie: {
+        name: 'language'
+      }
+    });
+    var req = getRequest({
+      acceptLanguage: 'en-US;q=1',
+      query: {
+        language: 'zh-CN'
+      }
+    });
+
+    middleware(req, res, next);
+    expect(req.cookies.language).to.equal('zh-CN');
+  });
+
+  it('should be able to use a custom query string name', function() {
+    var middleware = requestLangauge({
+      languages: ['en-US', 'zh-CN'],
+      queryName: 'locale',
+      cookie: {
+        name: 'language'
+      }
+    });
+    var req = getRequest({
+      acceptLanguage: 'en-US;q=1',
+      query: {
+        locale: 'zh-CN'
+      }
+    });
+
+    middleware(req, res, next);
+    expect(req.cookies.language).to.equal('zh-CN');
+  });
```

Comment:
Please also add a before mentioned test if the language is not supported. It should continue the normal logic chain(check cookie and if not check accept-lang header.).